docs/map.rst

```diff
@@ -0,0 +1,629 @@
+returns ``Nothing`` then the entry is deleted.
+
+
+Modifying all map entries (mapping)
+"""""""""""""""""""""""""""""""""""
+
+::
+
+    Map.map :: (a -> b) -> Map k a -> Map k v
+    Map.map f m = ...
+
+:map:`map` creates a new map by applying the transformation function ``f`` to
+each entries value. This is how `Functor
+<https://wiki.haskell.org/Typeclassopedia#Functor>`_ is defined for maps.
+
+::
+
+    Map.map (*10) (Map.fromList [("haskell", 45), ("idris", 15)])
+    > fromList [("haskell",450),("idris",150)]
+
+    -- Use the Functor instance for Map.
+    (*10) <$> Map.fromList [("haskell", 45), ("idris", 15)]
+    > fromList [("haskell",450),("idris",150)]
+
+There are several other more complex mapping functions available that let you
+look at other parts of the entry (such as they key) when transforming the
+value. For the full list see the `API documentation
+<https://hackage.haskell.org/package/containers-0.5.10.2/docs/Data-Map-Strict.html#g:15>`_.
+
+
+Set-like Operations
+^^^^^^^^^^^^^^^^^^^
+
+.. _union:
+
+Union
+"""""
+
+::
+
+    Map.union :: Ord k => Map k v -> Map k v -> Map k v
+    Map.union l r = ...
+
+:map:`union` returns a map containing all entries that are keyed in either of
+the two map. If the same key appears in both maps, the value from the left map
+``l`` taken (`set union <https://en.wikipedia.org/wiki/Union_(set_theory)>`_).
+
+::
+
+    Map.union Map.empty (Map.fromList [(1,"one"),(2,"two")])
+    > fromList [(1,"one"),(2,"two")]
+
+    Map.union (Map.fromList [(1, "uno")]) (Map.fromList [(1,"one"),(2,"two")])
+    > fromList [(1,"uno"),(2,"two")]
+
+Intersection
+""""""""""""
+
+::
+
+    Map.intersection :: Ord k => Map k v -> Map k v -> Map k v
+    Map.intersection l r = ...
+
+:map:`intersection` returns a map containing all entries that are keyed in both
+maps ``l`` and ``r``. The value from the left map is taken if the key exists in
+both maps (`set intersection
+<https://en.wikipedia.org/wiki/Intersection_(set_theory)>`_).
+
+::
+
+    Map.intersection Map.empty (Map.fromList [(1,"one"), (2,"two")])
+    > fromList []
+
+    Map.intersection (Map.fromList [(1, "uno")]) (Map.fromList [(1,"one"),(2,"two")])
+    > fromList [(1,"uno")]
+
+Difference
+""""""""""
+
+::
+
+    Map.difference :: Ord k => Map k v -> Map k v -> Map k v
+    Map.difference l r = ...
+
+:map:`difference` returns a map containing all entries that are keyed in the
+``l`` map but not the ``r`` map (`set difference/relative compliment
+<https://en.wikipedia.org/wiki/Complement_(set_theory)#Relative_complement>`_).
+
+::
+
+    Map.difference (Map.fromList [(1,"one"), (2,"two"), (3,"three")]) Map.empty
+    > fromList [(1,"uno"),(2,"two"),(3,"three")]
+
+    Map.difference (Map.fromList[(1,"one"), (2,"two")]) (Map.fromList [(1,"uno")])
+    > fromList [(2,"two")]
+
+Subset (submap)
+"""""""""""""""
+
+::
+
+    Map.isSubmapOf :: (Eq a, Ord k) => Map k v -> Map k v -> Bool
+    Map.isSubmapOf l r = ...
```

Comment:
I kind of doubt we should include this. It's a fairly unusual thing to want to do.

Reply:
Agreed, removed.